### PR TITLE
FIX Change timestamp type to 'ISO8601'

### LINKED ISSUE #265: Change timestamp type to 'ISO8601'
Current timestamp's attribute type is "timestamp". It should be changed to "ISO8601".

### PR DESCRIPTION
**Pull request by dmoranj**
Fixes #265

### CODE REVIEW

File: test/unit/compress-timestamp-plugin_test.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ describe('Timestamp compression plugin', function() {
             },
             {
                 name: 'The Target Value',
-                type: 'timestamp',
+                type: 'ISO8601',
                 value: '20071103T131805'
```
Comment (AlvaroVega) [inline test/unit/compress-timestamp-plugin_test.js]: is this value a right time ISO8601 ?

File: test/unit/contextRequests/updateContextCompressTimestamp1.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
         },
         {
           "name": "The Target Value",
-          "type": "timestamp",
+          "type": "ISO8601",
           "value": "+002007-11-03T13:18:05"
```
Comment (AlvaroVega) [inline test/unit/contextRequests/updateContextCompressTimestamp1.json]: I miss Timezone in value:```+002007-11-03T13:18:05Z```

File: test/unit/compress-timestamp-plugin_test.js (hunk 1 above)
Comment (dmoranj) [inline test/unit/compress-timestamp-plugin_test.js]: Quoting from Wikipedia:
```
Alternatively, a format for duration based on combined date and time representations may be used by agreement between the communicating parties either in the basic format PYYYYMMDDThhmmss or in the extended format P[YYYY]-[MM]-[DD]T[hh]:[mm]:[ss].
```
Maybe the P is missing is that's what your refer to. This format was the compressed one agreed with the Thinking Things for its use inside the devices. The IoTAgent itself transforms it to the expanded representation using the time compression plugin.

**Comment by AlvaroVega** [comment]
LGTM

**Pull request merged**